contentcuration/contentcuration/frontend/channelEdit/components/Clipboard/index.vue

```diff
@@ -248,13 +249,19 @@
       topicAndResourceCount() {
         let topicCount = 0;
         let resourceCount = 0;
+        const contentNodesValues = Object.values(this.contentNodesMap);
         this.selectedNodeIds.forEach(id => {
-          const kind = this.clipboardNodesMap[id]?.kind;
-          if (kind === 'topic' || !kind) {
-            // Increment topicCount if kind is "topic" or not set
+          const node = this.clipboardNodesMap[id];
+          let kind = node?.kind;
+          // Check contentNodesMap for node kind if missing in clipboardNodesMap
+          if (!kind && node?.source_node_id) {
+            const contentNode = contentNodesValues.find(n => n.node_id === node.source_node_id);
+            kind = contentNode?.kind;
```

Comment:
Non-blocking question: did you find that the `?.` was necessary here because `kind` and/or `source_node_id` was missing on some contents or was it to just avoid unhandled errors just in case?

I ask because if we _should_ expect `kind` on every `contentNode` (which I _think_ we should?) then using the `?.` might result in hard-to-debug issues when it isn't because it'll just count broken nodes as "resources" in any case here.

It's not crucial here, but in the if/else below it might be worth making the second condition like this (shortened for GH formatting).

```javascript
if(kind === 'topic) { topicCount++  }
else if (kind) { resourceCount++ }  // it's not undefined or null, so we know it's a non-topic resource
else { console.error("`kind` missing from content...")  }
```

But, again, this only really matters if we _should_ be able to expect `kind` to _always_ be defined in this context -- which may not be the case?

Reply:
Yes, we should be able to expect `kind` on every `contentNode`, the `?` is in place to avoid any potential errors, but you are right about potential difficulties with debugging so I will make this change.